server/middlewares/userValidation.js

```diff
@@ -1,21 +1,37 @@
 import Joi from '@hapi/joi';
 import userSignupSchema from '../schemas/userSignup';
+import resetPasswordSchema from '../schemas/resetPassword';
 import { validateInputs } from '../helpers';
 
+const { email, newPassword } = resetPasswordSchema;
+
 /**
- * Validates user registration/signup input
+ * Validates All User Inputs
  *
- * @param {string} req - ExpressJs request object
- * @param {string} res - ExpressJs response object
- * @param {string} next - ExpressJs next function
+ * @param {object} schema - ExpressJs request object
  * @returns {(JSON|function)} HTTP JSON response or ExpressJs next function
  */
-const validateUserSignup = (req, res, next) => {
-  const options = {
-    abortEarly: false
-  };
+const validate = (schema) => {
```

Comment:
Could you consider passing the express middleware functions to this validate function as well so that we have one validation function and not two?

Reply:
I believe we will still have two functions accepting parameters, but the es6 syntax looks good.